lib/chef/knife/winrm_base.rb

```diff
@@ -0,0 +1,117 @@
+#
+# Author:: Seth Chisamore (<schisamo@chef.io>)
+# Copyright:: Copyright (c) Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../knife"
+
+class Chef
+  class Knife
+    module WinrmBase
+
+      # It includes supported WinRM authentication protocol.
+      WINRM_AUTH_PROTOCOL_LIST ||= %w{basic negotiate kerberos}.freeze
+
+      # :nodoc:
+      # Would prefer to do this in a rational way, but can't be done b/c of
+      # Mixlib::CLI's design :(
+      def self.included(includer)
+        includer.class_eval do
+
+          deps do
+            require_relative "../encrypted_data_bag_item"
+            require "kconv"
+            require "readline"
+            require_relative "../json_compat"
+          end
+
+          option :winrm_user,
+            short: "-x USERNAME",
+            long: "--winrm-user USERNAME",
+            description: "The WinRM username",
+            default: "Administrator"
+
+          option :winrm_password,
+            short: "-P PASSWORD",
+            long: "--winrm-password PASSWORD",
+            description: "The WinRM password"
+
+          option :winrm_shell,
+            long: "--winrm-shell SHELL",
+            description: "The WinRM shell type. Valid choices are [cmd, powershell, elevated]. 'elevated' runs powershell in a scheduled task",
+            default: :cmd,
+            proc: Proc.new { |shell| shell.to_sym }
+
+          option :winrm_transport,
+            short: "-w TRANSPORT",
+            long: "--winrm-transport TRANSPORT",
+            description: "The WinRM transport type. Valid choices are [ssl, plaintext]",
```

Comment:
Same here, these valid choices should be an `in:` property.

Reply:
Fixed